lib/HLSL/DxilCondenseResources.cpp

```diff
@@ -1633,8 +1637,10 @@ Type *UpdateFieldTypeForLegacyLayout(Type *Ty,
         UpdateFieldTypeForLegacyLayout(EltTy, annotation, TypeSys, M);
     if (EltTy == UpdatedTy)
       return Ty;
-    else
+    else if (UpdatedTy)
       return ArrayType::get(UpdatedTy, Ty->getArrayNumElements());
+    else
+      return nullptr;
```

Comment:
`nullptr` is now used to indicate that the type is a resource or struct that's empty besides resources, and a field of this type should not be included in a translated structure.

Reply:
Maybe add this comment to the code?